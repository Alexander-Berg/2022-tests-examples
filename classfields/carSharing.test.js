const preparer = require('./carSharing');

let isCardPage;

describe('карточка:', () => {
    beforeEach(() => {
        isCardPage = true;
    });

    it('status === ERROR: вернет правильный результат', () => {
        const item = { status: 'ERROR' };
        const data = preparer({ item, isCardPage });

        expect(data).toMatchSnapshot();
    });

    it('status === UNKNOWN: вернет правильный результат', () => {
        const item = { status: 'UNKNOWN' };
        const data = preparer({ item, isCardPage });

        expect(data).toMatchSnapshot();
    });

    it('status === OK: вернет правильный результат', () => {
        const item = { status: 'OK' };
        const data = preparer({ item, isCardPage });

        expect(data).toMatchSnapshot();
    });

    it('если есть подозрение, поменяет статус на UNKNOWN', () => {
        const result = { report: { car_sharing: { car_sharing_records: 0, could_be_used_in_car_sharing: true } } };
        const item = { status: 'OK' };
        const data = preparer({ item, isCardPage, result });

        expect(data).toMatchSnapshot();
    });
});

describe('отчет:', () => {
    beforeEach(() => {
        isCardPage = false;
    });

    it('status === ERROR: вернет правильный результат', () => {
        const item = { status: 'ERROR' };
        const data = preparer({ item, isCardPage });

        expect(data).toMatchSnapshot();
    });

    it('status === UNKNOWN: вернет правильный результат', () => {
        const item = { status: 'UNKNOWN' };
        const data = preparer({ item, isCardPage });

        expect(data).toMatchSnapshot();
    });

    it('status === OK: вернет правильный результат', () => {
        const item = { status: 'OK' };
        const result = { report: { offer_id: '123' } };
        const data = preparer({ item, isCardPage, result });

        expect(data).toMatchSnapshot();
    });

    it('если есть подозрение, поменяет статус на UNKNOWN', () => {
        const result = { report: { car_sharing: { car_sharing_records: 0, could_be_used_in_car_sharing: true } } };
        const item = { status: 'OK' };
        const data = preparer({ item, isCardPage, result });

        expect(data).toMatchSnapshot();
    });
});
