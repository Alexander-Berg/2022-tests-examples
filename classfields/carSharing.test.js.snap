// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`карточка: status === ERROR: вернет правильный результат 1`] = `
Object {
  "key": "Работал в каршеринге",
}
`;

exports[`карточка: status === OK: вернет правильный результат 1`] = `
Object {
  "key": "Не работал в каршеринге",
}
`;

exports[`карточка: status === UNKNOWN: вернет правильный результат 1`] = `
Object {
  "key": "Возможно, работал в каршеринге",
}
`;

exports[`карточка: если есть подозрение, поменяет статус на UNKNOWN 1`] = `
Object {
  "key": "Возможно, работал в каршеринге",
  "status": "UNKNOWN",
}
`;

exports[`отчет: status === ERROR: вернет правильный результат 1`] = `
Object {
  "key": "Работа в каршеринге",
  "value": "Нет записей",
}
`;

exports[`отчет: status === OK: вернет правильный результат 1`] = `
Object {
  "key": "Работа в каршеринге",
  "value": "Нет записей",
}
`;

exports[`отчет: status === UNKNOWN: вернет правильный результат 1`] = `
Object {
  "key": "Работа в каршеринге",
  "value": "Есть подозрение",
}
`;

exports[`отчет: если есть подозрение, поменяет статус на UNKNOWN 1`] = `
Object {
  "key": "Работа в каршеринге",
  "status": "UNKNOWN",
  "value": "Есть подозрение",
}
`;
